modules/mfa/public/send-recovery-mfa.php

```diff
@@ -0,0 +1,55 @@
+<?php
```

Comment:
If you'd like to use the object-based controller mechanism that's new to SimpleSAMLphp 2.0, you can use the sildisco module as an example (or https://github.com/simplesamlphp/simplesamlphp/tree/master/modules/saml for a larger example). Actually, I'm not sure if it works to only add new actions to the class while leaving the existing ones in ordinary php files. But I thought I'd at least mention it in case you're interested.

I never found actual documentation on it. Maybe Symfony docs would be applicable. But the ultra-short TLDR is to create a route file in module/mfa/routing/routes/routes.yml and specify class methods in a class defined in module/mfa/src/Controller/Recovery.php

Reply:
Thanks! I don't think I will switch at this point, but I'm glad to know about that.